Code/Framework/AzCore/AzCore/DOM/DomPath.cpp

```diff
@@ -135,6 +135,58 @@ namespace AZ::Dom
             },
             m_value);
     }
+
+    bool operator<(const AZ::Dom::PathEntry& lhs, const AZ::Dom::PathEntry& rhs)
+    {
+        if (lhs.IsIndex())
+        {
+            return lhs.GetIndex() < rhs;
+        }
+        else
+        {
+            return lhs.GetKey() < rhs;
+        }
+    }
+
+    bool operator<(const AZ::Dom::PathEntry& entry, size_t index)
+    {
+        return entry.IsIndex() && entry.GetIndex() < index;
+    }
+
+    bool operator<(size_t index, const AZ::Dom::PathEntry& entry)
+    {
+        return !entry.IsIndex() || index < entry.GetIndex();
+    }
```

Comment:
The operation being performed here doesn't make any sense.

If the path entry isn't an index then the less than operator should be false in all cases

Reply:
`!entry.IsIndex() || index < entry.GetIndex()`
I was thinking it meant "return true if entry is a key, or return their comparison result if they are both indices".

My understanding is that we want non-index path entry values to be sorted after index path entries. Something like this:
```
/0
/1
/1/3
/0hello (because this is actually a key, not a size_t index)
/two
/two/three
```

With the code above, the comparison `1 < PathEntry("/hello")` would yield true because 1 is an index value and "hello" is a key value. So our sort order is:
```
/1
/hello
```

If we always return false when the path entry is not an index, then the sort order becomes:
```
/hello
/1
```

I should add an example sort to the description of this PR too so it's less unclear. And probably a comment inline somewhere.
@alexmontAmazon let me know if I misunderstood the desired sort order!

Reply:
I think it's reasonable to sort index entries before string entries, just so long as it's documented as such.

Reply:
But in JSON indices and strings aren't treated differently.
They sort together as an index is just treated as a string.
And by string sorting logic the hex values of 0x30 - 0x39 [0-9] sorts before 0x41 - 0x5a [A-Z]